### PR TITLE
Feature/4219 set password component

### PR DESCRIPTION
**Pull request by surajeaton**
Fixes # BLUI-4219

#### Changes proposed in this Pull Request:

- Added set password component
- Added example in example 2
- **Removed header description, as it makes more sense to be a part of screen which uses SetPassword component**

#### Screenshots / Screen Recording (if applicable)
Screenshot 2023-05-26 at 1 44 36 PM

#### To Test:

- yarn start:example2
- Click on Set Password button

### CODE REVIEW

**Comment by JeffGreiner-eaton** [comment]
I know we mentioned a couple questions for this in stand-up today but if this screen is part of the registration workflow shouldn't it be the same as it is in dev? @daileytj @surajeaton 

image

**Comment by JeffGreiner-eaton** [comment]
Without any input change and onBlur we throw input error
image

**Comment by JeffGreiner-eaton** [comment]
question - do we want this random failure to remain as is like it was implemented in the old version? what is the purpose of it for a user and will the user be able to adjust this? @daileytj @surajeaton 

image

**Comment by daileytj** [comment]
> question - do we want this random failure to remain as is like it was implemented in the old version? what is the purpose of it for a user and will the user be able to adjust this? @daileytj @surajeaton
> 
This is a method for us to test random failures in the example project. I don't think we need this utility right now.

**Comment by surajeaton** [comment]
> Without any input change and onBlur we throw input error image

Yes, this functionality is correct and same is there in dev.

**Comment by surajeaton** [comment]
> I know we mentioned a couple questions for this in stand-up today but if this screen is part of the registration workflow shouldn't it be the same as it is in dev? @daileytj @surajeaton
> 
> Dev:
> 
> image

SetPassword component is just rendering the password, confirm password fields and password requirements.
The header (line 1) and instruction (line 2) needs to be set up in the screen which uses SetPassword inside. The Screen will be implemented in the another story which Ekta is working on (CreatePasswordScreenBase)

**Comment by surajeaton** [comment]
@JeffGreiner-eaton Please review.

**Comment by JeffGreiner-eaton** [comment]
> > Without any input change and onBlur we throw input error image
> 
> Yes, this functionality is correct and same is there in dev.

This is dev and when input field is not dirty we don't throw error...


https://github.com/etn-ccis/blui-react-workflows/assets/119693939/067d572e-27dc-4133-bf40-55938aaf962b

File: login-workflow/example2/src/screens/new-architecture-test-screens/SetPasswordScreen.tsx [hunk 1]
```diff
@@ -0,0 +1,76 @@
+import React, { useState, useCallback, useEffect, useRef } from 'react';
+import {
+    defaultPasswordRequirements,
+    SetPassword,
+    useLanguageLocale,
+    AuthUIContextProvider,
+    useSecurityActions,
+    BrandedCardContainer,
```
Comment (JeffGreiner-eaton) [inline login-workflow/example2/src/screens/new-architecture-test-screens/SetPasswordScreen.tsx:8]: This file looks like our original, why are we using branded card container, t function on lang. etc. @daileytj  what are the plans here? Will this be refactored later?

**Comment by JeffGreiner-eaton** [comment]
The password component itself seems good. There are other integrations in here I'm not familiar enough with to give approval on and waiting for comments from @daileytj

File: login-workflow/example2/src/screens/new-architecture-test-screens/SetPasswordScreen.tsx (hunk 1 above)
Comment (daileytj) [reply login-workflow/example2/src/screens/new-architecture-test-screens/SetPasswordScreen.tsx:8]: Anything that is used should be used from the new-architecture folder. If there are items we are using that need migrated then migrate them. In the case of BrandedCardContainer, we'll need to have that replaced with the new WorkflowCard components

Comment (surajeaton) [reply login-workflow/example2/src/screens/new-architecture-test-screens/SetPasswordScreen.tsx]: Done. We are using everything from new-architecture and replaced BrandedCardContainer with WorkflowCard. Please review.

**Comment by surajeaton** [comment]
I have added the test cases in the same PR.

**Review by JeffGreiner-eaton — APPROVED** [review]
this looks good to me, test coverage good as well

**Pull request merged**